crates/holochain/tests/app_interface_security.rs

```diff
@@ -0,0 +1,417 @@
+use either::Either;
+
+use holochain::sweettest::{websocket_client_by_port, SweetConductor, SweetDnaFile, WsPollRecv};
+use holochain_conductor_api::{
+    AdminRequest, AdminResponse, AppAuthenticationRequest, AppAuthenticationToken, AppRequest,
+    AppResponse,
+};
+use holochain_types::prelude::InstalledAppId;
+use holochain_types::websocket::AllowedOrigins;
+use holochain_wasm_test_utils::TestWasm;
+use holochain_websocket::ReceiveMessage;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn app_interface_requires_auth() {
+    holochain_trace::test_run();
+
+    let conductor = SweetConductor::from_standard_config().await;
+
+    // App interface with no restrictions, but should still require auth
+    let app_port = conductor
+        .clone()
+        .add_app_interface(Either::Left(0), AllowedOrigins::Any, None)
+        .await
+        .unwrap();
+
+    let (app_tx, app_rx) = websocket_client_by_port(app_port).await.unwrap();
+    let _app_rx = WsPollRecv::new::<AppResponse>(app_rx);
+
+    // Try to send a request before authenticating, results in connection closed
+    let err = app_tx
+        .request::<_, AppResponse>(AppRequest::AppInfo)
+        .await
+        .unwrap_err();
+    assert_eq!("ConnectionClosed", err.to_string());
+
+    let token = create_token(&conductor, "test-app".into()).await;
+
+    // Try to authenticate against the connection which is supposed to be closed
+    let err = app_tx
+        .authenticate(AppAuthenticationRequest {
+            token: token.clone(),
+        })
+        .await
+        .unwrap_err();
+    assert_eq!("WebsocketClosed", err.to_string());
+
+    // Token didn't get used above, so create a new connection and try to use it
+    let (app_tx, app_rx) = websocket_client_by_port(app_port).await.unwrap();
+    let _app_rx = WsPollRecv::new::<AppResponse>(app_rx);
+
+    app_tx
+        .authenticate(AppAuthenticationRequest {
+            token: token.clone(),
+        })
+        .await
+        .unwrap();
+
+    // Authentication should have worked, so now we can make requests
+    let response: AppResponse = app_tx
+        .request(AppRequest::ListWasmHostFunctions)
+        .await
+        .unwrap();
+    assert!(matches!(response, AppResponse::ListWasmHostFunctions(_)));
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn app_interface_can_handle_bad_auth_payload() {
```

Comment:
@jost-s This is the bad authentication payload test from the previous PR that I said I'd add